core/sail/nativerdf/src/test/java/org/eclipse/rdf4j/sail/nativerdf/NativeSailStoreCorruptionTest.java

```diff
@@ -0,0 +1,224 @@
+
+		// Use RandomAccessFile in "rw" mode to read and write to the file
+		try (RandomAccessFile raf = new RandomAccessFile(valuesFile, "rw")) {
+			// Get the length of the file
+			long fileLength = raf.length();
+
+			// Check if the position is within the file bounds
+			if (pos >= fileLength) {
+				throw new IOException(
+						"Attempt to write outside the existing file bounds: " + pos + " >= " + fileLength);
+			}
+
+			// Move the file pointer to byte position 32
+			raf.seek(pos);
+
+			// Write the byte value 0x0 at the current position
+			raf.writeByte(newVal);
+		}
+	}
+
+	public static void backupFile(File dataDir, String s) throws IOException {
+		File valuesFile = new File(dataDir, s);
+		File backupFile = new File(dataDir, s + ".bak");
+
+		if (!valuesFile.exists()) {
+			throw new IOException("values.dat does not exist and cannot be backed up.");
+		}
+
+		// Copy values.dat to values.dat.bak
+		Files.copy(valuesFile.toPath(), backupFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+	}
+
+	public static void restoreFile(File dataDir, String s) throws IOException {
+		File valuesFile = new File(dataDir, s);
+		File backupFile = new File(dataDir, s + ".bak");
+
+		if (!backupFile.exists()) {
+			throw new IOException("Backup file values.dat.bak does not exist.");
+		}
+
+		// Copy values.dat.bak back to values.dat
+		Files.copy(backupFile.toPath(), valuesFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+	}
+
+	@Test
+	public void testCorruptValuesDatFileNamespace() throws IOException {
+		repo.shutDown();
+
+		overwriteByteInFile(new File(dataDir, "values.dat"), 12, 0x0);
+
+		repo.init();
+
+		List<Statement> list = getStatements();
+		assertEquals(6, list.size());
+	}
+
+	@Test
+	public void testCorruptValuesDatFileNamespaceDatatype() throws IOException {
+		repo.shutDown();
+
+		overwriteByteInFile(new File(dataDir, "values.dat"), 96, 0x0);
+
+		repo.init();
+
+		List<Statement> list = getStatements();
+		assertEquals(6, list.size());
+	}
+
+	@Test
+	public void testCorruptValuesDatFileEmptyDataArrayError() throws IOException {
+		repo.shutDown();
+
+		overwriteByteInFile(new File(dataDir, "values.dat"), 173, 0x0);
+
+		repo.init();
+
+		List<Statement> list = getStatements();
+		assertEquals(6, list.size());
+	}
+
+	@Test
+	public void testCorruptValuesDatFileInvalidTypeError() throws IOException {
+		repo.shutDown();
+
+		overwriteByteInFile(new File(dataDir, "values.dat"), 174, 0x0);
+
+		repo.init();
+
+		List<Statement> list = getStatements();
+		assertEquals(6, list.size());
+	}
+
+	@Test
+	public void testCorruptValuesDatFileEntireValuesDatFile() throws IOException {
+		for (int i = 4; i < 437; i++) {
+			logger.debug("Corrupting byte at position " + i);
+			repo.shutDown();
+			restoreFile(dataDir, "values.dat");
+
+			overwriteByteInFile(new File(dataDir, "values.dat"), i, 0x0);
+
+			repo.init();
+
+			List<Statement> list = getStatements();
+			assertEquals(6, list.size());
+
+		}
+
+	}
```

Comment:
This is actually the main test that tests every single byte, except the header bytes, in the values.dat file.